### PR TITLE
Add a command line example for how to use libcgroup with systemd

### PR DESCRIPTION
**Pull request by drakenclimber**

### CODE REVIEW

File: src/api.c [hunk 1]
```diff
@@ -2919,8 +2924,20 @@ int cgroup_create_cgroup(struct cgroup *cgroup, int ignore_ownership)
...
+
+			/*
+			 * This will remove any cgroup directories that were made, but it won't
+			 * undo changes that have been written to the parent cgroup's
+			 * subtree_control file.  To safely undo changes there, we would need to
+			 * save the subtree_control file's previous value and restore it.
+			 */
+			del_error = cgroup_delete_cgroup(cgroup, 1);
+			if (del_error)
+				cgroup_err("Failed to delete %s: %s\n", cgroup->name, del_error);
```
Comment (kamalesh-babulal) [inline src/api.c:2938]: @drakenclimber The patch looks good to me. A minor nitpick, in the above error message the format string of `del_error` is of string, was the intention to print `cgroup_strerror(del_error))`

File: tests/ftests/086-sudo-systemd_cmdline_example.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    # Step 2 in the example
+    # $ sudo cgcreate -g cpu,memory:mycompany.slice/database.scope/high-priority
+    # cgcreate: can't create cgroup mycompany.slice/database.scope/high-priority: Operation not
+    # supported
+    #
+    try:
+        CgroupCli.create_and_validate(config, CONTROLLER_LIST, HIGH_CGNAME)
+    except RunError as re:
+        if re.ret != 96 or 'No such file or directory' not in re.stderr:
+            print(re)
```
Comment (kamalesh-babulal) [inline tests/ftests/086-sudo-systemd_cmdline_example.py:80]: Debugging statement :)

File: samples/cmdline/systemd-with-idle-process.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# Creating a Systemd Scope and Child Hierarchy via Libcgroup Command Line
```
Comment (kamalesh-babulal) [inline samples/cmdline/systemd-with-idle-process.md:1]: Pretty awesome documentation, thank you so much for writing it and also adding an example test program.

File: tests/ftests/086-sudo-systemd_cmdline_example.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    pids = CgroupCli.get_pids_in_cgroup(config, os.path.join(SLICE, SCOPE, TMP_CGNAME), CONTROLLER_LIST[0])
+    Process.kill(config, pids)
+
+    if result != consts.TEST_PASSED:
+        # Something went wrong.  Let's force the removal of the cgroups just to be safe.
+        # Note that this should remove the cgroup, but it won't remove it from systemd's
+        # internal caches, so the system may not return to its 'pristine' prior-to-this-test
+        # state
+        try:
+            CgroupCli.delete(config, None, os.path.join(SLICE, SCOPE))
```
[the remark this answers is not in the document]
Comment (drakenclimber) [reply tests/ftests/086-sudo-systemd_cmdline_example.py:167]: Good call.  Deleting the slice is the right move

File: src/api.c (hunk 1 above)
Comment (drakenclimber) [reply src/api.c:2938]: Good find.  Thanks

**Comment by drakenclimber** [comment]
I uploaded v2.  Changes include:
* address the code review comments
* fix lint warnings

**Review by kamalesh-babulal — APPROVED** [review]

**Comment by drakenclimber** [comment]
Pushed to main.  Thanks for the help, @kamalesh-babulal

**Pull request closed without merging**